src/api/java/Optimize.java

```diff
@@ -150,19 +150,19 @@ public String toString()
      * Return an objective which associates with the group of constraints.
      *
      **/
-    public Handle AssertSoft(BoolExpr constraint, int weight, String group)
+    public Handle<?> AssertSoft(BoolExpr constraint, int weight, String group)
```

Comment:
Requires explicit casting by user because there's no input to base the result type with; I think this is a reasonable requirement on the user. Won't break existing code.